components/automate-ui/src/app/page-components/insight-attributes-dropdown/insight-attributes-dropdown.component.spec.ts

```diff
@@ -0,0 +1,131 @@
+      const buttonsArray = fixture.debugElement.queryAll(By.css('.filter-button'));
+
+      expect(buttonsArray.length).toBe(component.options.length);
+    });
+
+    it('the update button is disabled', () => {
+      const updateButton = fixture.debugElement.nativeElement.querySelector('chef-button[primary]');
+
+      expect(updateButton.disabled).toBeTruthy();
+    });
+  });
+
+
+  describe('when making selections to filter', () => {
+    beforeEach(() => {
+      component.options = testOptions;
+      fixture.detectChanges();
+    });
+
+    it('adds the selected buttons id to the selectedOptions array', () => {
+      const list = fixture.debugElement.nativeElement.querySelectorAll('.filter-button');
+      list[0].click();
+      list[1].click();
+      list[4].click();
+
+      expect(component.selectedOptions.length).toBe(3);
+      expect(component.selectedOptions).toContain('platform_filter');
+      expect(component.selectedOptions).toContain('domain_filter');
+      expect(component.selectedOptions).toContain('hostname_filter');
+      expect(component.selectedOptions).not.toContain('macaddress_filter');
+      expect(component.selectedOptions).not.toContain('cloud_provider_filter');
+    });
+
+    it('allows a maximum of 5 selected options', () => {
+      const list = fixture.debugElement.nativeElement.querySelectorAll('.filter-button');
+      expect(list.length).toBe(component.options.length);
+
+      list.forEach(option => option.click());
+
+      expect(component.selectedOptions.length).toBe(5);
+    });
+
+    it('enables the submit when filters are different than previous selection', () => {
+      component.selectedOptions = ['platform_filter', 'domain_filter'];
+      component.lastSelectedOptions = ['platform_filter', 'domain_filter'];
+      const updateButton = fixture.debugElement.nativeElement.querySelector('chef-button[primary]');
+      const list = fixture.debugElement.nativeElement.querySelectorAll('.filter-button');
+
+      expect(updateButton.disabled).toBeTruthy();
+
+      list[0].click();
+      fixture.detectChanges();
+      expect(updateButton.disabled).toBeFalsy();
+
+      list[4].click();
+      fixture.detectChanges();
+      expect(updateButton.disabled).toBeFalsy();
+
+      list[0].click();
+      list[4].click();
+      fixture.detectChanges();
+      expect(updateButton.disabled).toBeTruthy();
+    });
+
+    it('updates selected class and aria-pressed value', () => {
+      const list = fixture.debugElement.nativeElement.querySelectorAll('.filter-button');
+      list[0].click();
+      list[1].click();
+      list[4].click();
+      fixture.detectChanges();
+
+      const allSelected = fixture.debugElement.nativeElement
+        .querySelectorAll('.filter-button.selected');
+      const allPressed = fixture.debugElement.nativeElement
+        .querySelectorAll('[aria-pressed="true"]');
+
+      expect(allSelected.length).toBe(3);
+      expect(allPressed.length).toBe(3);
+    });
+  });
+
```

Comment:
I don't yet have testes for submitted/cancel/or checkbox because it is almost a guarantee that those responsibilities are going to change anyways.